web/src/components/tasks/EinInput.tsx

```diff
@@ -21,24 +22,29 @@ export const EinInput = (props: Props): ReactElement => {
   const [isInvalid, setIsInvalid] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [employerId, setEmployerId] = useState<string>("");
+  const { isAuthenticated, setRegistrationModalIsVisible } = useContext(AuthAlertContext);
   const { business, updateQueue } = useUserData();
   const { queueUpdateTaskProgress } = useUpdateTaskProgress();
 
-  let saveButtonText = Config.ein.saveButtonText;
-  if (props.isAuthenticated === IsAuthenticated.FALSE) {
-    saveButtonText = `Register & ${saveButtonText}`;
-  }
-
   useMountEffectWhenDefined(() => {
     if (!business) return;
     setEmployerId(business.profileData.employerId || "");
   }, business);
 
   const handleChange = (value: string): void => {
+    if (isAuthenticated === IsAuthenticated.FALSE) {
+      setRegistrationModalIsVisible(true);
+      return;
+    }
     setEmployerId(value);
   };
 
   const save = async (): Promise<void> => {
+    if (isAuthenticated === IsAuthenticated.FALSE) {
+      setRegistrationModalIsVisible(true);
+      return;
+    }
```

Comment:
should we put this in a separate function since this is repeated above?

Reply:
I tried doing that but we need the return statement to exit the function so even after extraction it ends up being 
```
if(unauthenticated){
 functionToOpen();
 return
}
```
I could rewrite the logic so the it's an `if/else` but even then the function would still just be opening the modal? Do you have any suggestions how to reduce the repetition?

Reply:
ok i am with anne